### PR TITLE
rework landing docs

### PR DESCRIPTION
**Pull request by yoshuawuyts**
The current docs are a bit much. This is an attempt to be more to the point. More showing, less telling. Thanks!

__edit:__ all the how-to's should move to their own guides. Planning to do that over the course of next week.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,20 +1,13 @@
...
-//! # Features
-//!
-//! - __Fast:__ Written in Rust, and built on Futures, Tide is incredibly efficient.
-//! - __Friendly:__ With thorough documentation, and a complete API, Tide helps cover your every
-//!     need.
-//! - __Minimal:__ With only a few concepts to learn, Tide is easy to pick up and become productive
-//!     with.
+//! Tide is a minimal and pragmatic Rust web application framework built for
+//! rapid development. It comes with a robust set of features that make
+//! building database-backed web applications easier and more fun.
```
Comment (Fishrock123) [inline src/lib.rs:3]: Can we update the `Cargo.toml` description with this too?

File: README.md [hunk 2]
```diff
@@ -41,11 +41,16 @@
   </h3>
 </div>
 
-A modular web framework built around async/await
+Tide is a minimal and pragmatic Rust web application framework built for
+rapid development. It comes with a robust set of features that make building
+database-backed web applications easier and more fun.
```
Comment (jbr) [inline README.md:46]: Database-backed seems more specific/constrained than tide is? Suggestion: "that make building web applications and apis easier and more fun."

File: README.md [hunk 3]
```diff
@@ -54,25 +59,43 @@ async-std = { version = "1.6.0", features = ["attributes"] }
 
 ## Examples
 
-**Hello World**
+Create an HTTP server that takes JSON, validates it, and responds with a
```
Comment (jbr) [inline README.md:62]:
```suggestion
Create an HTTP server that receives JSON, validates it, and responds with a
```

File: src/lib.rs [hunk 4]
```diff
@@ -1,197 +1,56 @@
...
```
Comment (jbr) [inline src/lib.rs:3]: same comment here as in cargo.toml — not all tide apps are database-backed, nor does tide really do anything special to enable database-backed apps

File: src/lib.rs [hunk 5]
```diff
@@ -1,197 +1,56 @@
...
-//! # use async_std::task::block_on;
-//! # fn main() -> Result<(), std::io::Error> { block_on(async {
-//! #
-//! let mut app = tide::new();
-//! app.at("/").get(|_| async { Ok("Hello, world!") });
-//! app.listen("127.0.0.1:8080").await?;
-//! #
-//! # Ok(()) }) }
-//! ```
+//! Create an HTTP server that takes JSON, validates it, and responds with a
```
Comment (jbr) [inline src/lib.rs:19]:
```suggestion
//! Create an HTTP server that receives a JSON body, validates it, and responds with a
```

**Review by jbr — APPROVED** [review]
This looks like a great improvement for the new user experience! Some small tweaks:

File: README.md (hunk 2 above)
Comment (Fishrock123) [reply README.md:46]: I think it may still be worth mentioning futures/async in the description as well.

**Pull request merged**